assets/js/modules/analytics/datastore/report.js

```diff
@@ -120,6 +121,53 @@ const fetchGetReportStore = createFetchStore( {
 	},
 } );
 
+const gatheringDataStore = createGatheringDataStore( 'analytics', {
+	storeName: MODULES_ANALYTICS,
+	dataAvailable:
+		global._googlesitekitModulesData?.[ 'data_available_analytics' ],
+	selectDataAvailability: createRegistrySelector( ( select ) => () => {
+		const { startDate, endDate } = select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+		} );
+
+		const args = {
+			dimensions: [ 'ga:date' ],
+			metrics: [ { expression: 'ga:users' } ],
+			startDate,
+			endDate,
+		};
+
+		const url = select( CORE_SITE ).getCurrentEntityURL();
+		if ( url ) {
+			args.url = url;
+		}
+
+		// Disable reason: select needs to be called here or it will never run.
+		// eslint-disable-next-line @wordpress/no-unused-vars-before-return
+		const report = select( MODULES_ANALYTICS ).getReport( args );
+		const hasResolvedReport = select(
+			MODULES_ANALYTICS
+		).hasFinishedResolution( 'getReport', [ args ] );
+
+		if ( ! hasResolvedReport ) {
+			return undefined;
+		}
+
+		if ( ! Array.isArray( report ) ) {
+			return true;
+		}
```

Comment:
I know we negated all the returns here but this seems odd 🤔

Reply:
Yep looks pretty odd, initially I refactored this part to be cleaner but it kept getting lost in subsequent develop merges. So I just inverted the returns here. :sweat_smile: